lib/kubeclient/common.rb

```diff
@@ -477,5 +482,16 @@ def http_options(uri)
 
       options.merge(@socket_options)
     end
+
+    def check_token_expiration(http_code, message)
+      if http_code == 401 &&
+         @auth_options.key?(:bearer_token_expiry) &&
+         @auth_options[:bearer_token_expiry] < Time.now
+        return message + '. Your token has expired. If you are using `kubectl`, running any '\
+          'command should renew your token.'
```

Comment:
Maybe I don't understand this but it does not seem like a flow that a normal Kubeclient user would use.
(use library to talk to api while using command line tool to get authentication token). Can we do the same thing kubectl does to renew the token? what is needed for the kubectl to give back a token (meaning do you have to be pre authenticated against www.googleapis.com/auth/cloud-platform ?) 

This seems relevant: https://github.com/kubernetes/kubernetes/blob/3e51ca2e16479763bc832e893e1b9ffa34310b7e/plugin/pkg/client/auth/gcp/gcp.go#L56